beets/util/hidden.py

```diff
@@ -23,13 +23,15 @@
 from typing import Union
 
 
-def is_hidden(path: Union[bytes, Path]) -> bool:
+def is_hidden(path: Union[bytes, str, Path]) -> bool:
     """
     Determine whether the given path is treated as a 'hidden file' by the OS.
     """
 
     if isinstance(path, bytes):
         path = Path(os.fsdecode(path))
+    elif isinstance(path, str):
+        path = Path(path)
```

Comment:
Especially about this change I'm not sure. `is_hidden` is only called here throughout the codebase:

https://github.com/beetbox/beets/blob/5d010e95ec8940d59aac9f9c0dd987a77f617dd8/beets/util/__init__.py#L248

It is made sure that paths are wrapped using the `bytestring_path` util. So shouldn't we better fix `test_is_hidden` to wrap using `bytestring_path` rather than accepting `str` here?

Reply:
Yes absolutely. As I just re-discovered, there's a [section](https://github.com/beetbox/beets/blob/master/CONTRIBUTING.rst#handling-paths) in CONTRIBUTING.rst about it. I made that change.